classes/class.pmproemail.php

```diff
@@ -1306,26 +1306,70 @@ function sendTrialEndingEmail( $user = NULL, $membership_id = NULL )
 			return $this->sendEmail();
 		}
 		
-		
-		function sendMembershipExpiredEmail( $user = NULL, $membership_id = NULL )
-		{
+		/**
+		 * Send the member an email when their membership has ended.
+		 *
+		 * @param object $user The WordPress user object.
+		 * @param int $membership_id The member's membership level ID.
+		 * @return bool Whether the email was sent successfully.
+		 * @since TBD
+		 */
+		function sendMembershipExpiredEmail( $user = NULL, $membership_id = NULL ) {
 			global $current_user;
-			if(!$user)
+			if( !$user ) {
 				$user = $current_user;
-			
-			if(!$user)
-				return false;						
+			}
+			//Bail if still we don't have a user.
+			if( !$user ) {
+				return false;
+			}
+			//If we have a membership ID, get the level.
+			if (! empty( $membership_id ) ) {
+				$membership_level = pmpro_getLevel( $membership_id );
+			//If not let's find out what's the last ended membership level.
+			} else {
+				//Get all levels for a specific member
+				$all_levels = pmpro_getMembershipLevelsForUser( $user->ID );
```

Comment:
`false` can be passed as a second parameter to `pmpro_getMembershipLevelsForUser()` to get inactive levels in addition to the user's active records. If we are looking for the most recently ended membership, having inactive levels in the `$all_levels` array seems necessary.